### PR TITLE
Add map popup link to selected feature

### LINKED ISSUE #227: Make map popups lead to detail pages
Add a [chevron](https://mui.com/components/material-icons/?query=chevron&selected=ChevronRight) to MapPopupContent when it's expanded, like the popup in [this Figma mockup](https://www.figma.com/file/yaKUmuZM9xPQ1Qq5YVf5Yu/Design---Skills?node-id=3206%3A31143).

The title and chevron should be link to the detail page for the highlighted region. The link should be of the format `{baseURL}/explore/{dataset}/{id}`, e.g. Adelaide would be http://localhost:3000/explore/small-business/401011001.

Purple outline here to show the link's clickable area.

### PR DESCRIPTION
**Pull request by torosyaneric**
Resolves #227

### CODE REVIEW

**Review by justinanderson — CHANGES_REQUESTED** [review]
- [ ] The location hash shouldn't be kept when sending a user to a detail page. For example, I clicked on a region and got sent to http://localhost:3000/explore/small-business/407031164#map=7.03/-34.462/137.697 but should have been sent to http://localhost:3000/explore/small-business/407031164.

- [ ] I'm seeing a focus outline around the popup's title just from mousing around the map. The title link shouldn't gain focus unless a user tabs to it. I think it's stealing focus whenever the popup becomes visible. I think `mapboxgl.Popup`'s `options.focusAfterOpen` default value is causing that to happen.
Screen Shot 2022-03-28 at 6 07 57 PM

**Review by justinanderson — CHANGES_REQUESTED** [review]
- [ ] I just remembered that the chevron is supposed to be invisible when MapPopupContent isn't expanded. It should still take up space, just be invisible when not expanded, e.g. "Karoona Lameroo" should already wrap to two lines even when the popup isn't expanded.

File: frontend/src/components/Map/Map.js [hunk 1]
```diff
@@ -451,6 +452,9 @@ function Map({ config, hidePopup, datasetId }) {
     return () => {
       map.current?.remove?.()
       map.current = null
+      
+      // resets map's location hash
+      window.history.replaceState(null, null, ' ')
```
[the remark this answers is not in the document]
Comment (torosyaneric) [reply frontend/src/components/Map/Map.js:457]: @justinanderson can't find a fix. Shall I implement custom hash updater along with setting map coords on init?

**Comment by torosyaneric** [comment]
@justinanderson added popup chevron hiding.

File: frontend/src/components/Map/Map.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (justinanderson) [reply frontend/src/components/Map/Map.js:457]: Nah, maybe later. We don't have need yet on this page for links with hashes in them, so I'm inclined to leave this as a known issue for now. We can always make a custom hash updater later if that need changes.

**Review by justinanderson — APPROVED** [review]

**Pull request merged**